GitTfs.VsCommon/TfsHelper.Common.cs

```diff
@@ -618,8 +618,7 @@ public void WithWorkspace(string localDirectory, IGitTfsRemote remote, IEnumerab
             if (!_workspaces.TryGetValue(remote.Id, out workspace))
             {
                 Trace.WriteLine("Setting up a TFS workspace with subtrees at " + localDirectory);
-                var folders = mappings.Select(x => new WorkingFolder(x.Item1, Path.Combine(localDirectory, x.Item2))).ToArray();
-                _workspaces.Add(remote.Id, workspace = Retry.Do(() => GetWorkspace(folders)));
+                _workspaces.Add(remote.Id, workspace = Retry.Do(() => GetWorkspace(mappings.Select(x => new WorkingFolder(x.Item1, Path.Combine(localDirectory, x.Item2))).ToArray())));
```

Comment:
There was a change here. Sorry it was a sloppy change on my part. Expanded out to multi-line. 

Should be more clear now. Guarding in Retry.Do() if WorkingFolder errors.